### ISSUE TITLE
Fix false positives for nesting selector within at-includes in selector-no-qualifying-type

### ISSUE THREAD

**Issue by FRITS-Kh** [stylelint/stylelint#5020]
> Clearly describe the bug

Reports a wrong warning In SCSS.

> Which rule, if any, is the bug related to?

`selector-no-qualifying-type`

> What code is needed to reproduce the bug?

```scss
@mixin test($name: '.test') {
  #{$name} {
    &--test {
      // some styles
          
      &.active {
        display: none;
      }
    }

    @content;
  }
}

@include test {
  &--second {
    // some styles
        
    &.active {
      display: none;
    }
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-no-qualifying-type": true
  }
}
```

> Which version of stylelint are you using?

`13.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "assets/styles/**/*.scss"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS syntax.

> What did you expect to happen?

No warning for the example.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Unexpected qualifying type selector   selector-no-qualifying-type
```

Example of the issue online:
https://stylelint.io/demo#N4Igxg9gJgpiBcIACBbAlgDzQOwAQBcYBnfACgBJsBDFGeXAcgDpCSGBKXYAHT1wGJglGjAC+XXrim4AZAFo5rfBL7TcAenW4iEWtvwBPADbFJa82bUymVMPjQA3GCvPSoaIgAcjVA-WwQ2DAA3JbSomERqrhhSJDYhAmhfFFRvEg4YEYArrAExMo8fPJyRDDxUC7Smtq6ziTGptFqYda29k5Vau5ePn64AUHJalFSqSAANCBEBglUGAjTYEREk+CBAGZoAOaLRVLcIABO2SZEh-T70odlJnYQR3IBcgCO2VRGaBsGONuKBp4YBcCCcgSleKIQKIgA

**Comment by jeddy3** [comment]
@FRITS-Kh Thanks for the clear report, using the template and for providing a reproducible demo.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

I believe that:

```scss
@include test {
  &--second {
    // some styles
        
    &.active {
      display: none;
    }
  }
}
```

Is being resolved to:

```css
&--second.active {}
```

And the selector parser is treating `&--second` as a type selector. Hence, the rule thinks the `.active` class selector is qualified. 

I suggest we use the `isStandardSyntaxTypeSelector` util in the rule to exclude unresolved nesting selector masquerading as type selectors.
